File: backend/src/main/java/com/devmatch/backend/domain/application/repository/ApplicationRepository.java
package com.devmatch.backend.domain.application.repository;

import com.devmatch.backend.domain.application.entity.Application;
import com.devmatch.backend.domain.application.enums.ApplicationStatus;
import java.util.List;
import org.springframework.data.jpa.repository.JpaRepository;

public interface ApplicationRepository extends JpaRepository<Application, Long> {

  List<Application> findAllByUserId(Long id);

  List<Application> findAllByProjectId(Long id);

  List<Application> findByProjectIdAndStatus(Long projectId, ApplicationStatus status);

}

File: backend/src/main/java/com/devmatch/backend/domain/application/repository/ApplicationRepository.kt
package com.devmatch.backend.domain.application.repository

import com.devmatch.backend.domain.application.entity.Application
import com.devmatch.backend.domain.application.enums.ApplicationStatus
import org.springframework.data.jpa.repository.JpaRepository

interface ApplicationRepository : JpaRepository<Application, Long> {
    fun findAllByUserId(id: Long): List<Application>

    fun findAllByProjectId(id: Long): List<Application>

    fun findByProjectIdAndStatus(
        projectId: Long,
        status: ApplicationStatus
    ): List<Application>
}